app/src/main/java/org/eyeseetea/malariacare/data/net/ConnectivityManager.java

```diff
@@ -15,11 +15,14 @@
      * @return true if the device has a network connection, false otherwise.
      */
     @Override
-    public boolean isDeviceOnline() {
+    public boolean isDeviceOnline(boolean canDownloadWith3G) {
```

Comment:
we should not mix responsibilities, connectivity manager should only manage about network but it is not the responsibility of ConnectivityManager manage canDownloadWith3G. I think is better that connectivity manager returns type of connectivity and use case manage this information and canDownloadWith3G settings to decide. 
ConnectivityManager can have two methods isDeviceOnline and getTypeConnection for example.